tools/Replay.hpp

```diff
@@ -216,32 +267,104 @@ struct Stats {
         dumpMap(_recvd);
     }
 
-    void dumpStressToCSV(size_t runMs)
+
+    //loading phase will record the set up times and loaded phase records the document loading times
+    void perfSetupLoadEditPhases(std::string phase)
     {
-        std::ofstream file("CPU.csv", std::ios::out | std::ios::app);
+        const auto now = std::chrono::steady_clock::now();
+        const size_t runMs = std::chrono::duration_cast<std::chrono::milliseconds>(now - _start).count();
+        _start = std::chrono::steady_clock::now();
+
+        size_t cpuTime = _timer->elapsedTime().count();
+        _timer.reset(new Util::SysStopwatch);
 
-        if (file.tellp() == 0)
+        perfStatsList.push_back(GetStressStats(runMs, phase));
+        perfStatsList.push_back(GetCPUUSageStats(cpuTime, phase));
+
+        if(phase == "Edit")
         {
-            file << "Stress (m/s)";
-            file << "\n";
+            std::vector<PerfMetricInfo> statsList;
+
+            perfStatsList.push_back(GetPeakMemoryUsageStats(_peakMemoryUsage, phase));
+
+            statsList = GetNetworkStats(_bytesRecvd / 1000, _bytesSent / 1000, phase);
+            for(size_t i = 0; i < statsList.size(); i++)
+            {
+                perfStatsList.push_back(statsList[i]);
+            }
+
+            statsList = _pingLatency.GetLatencyStats("PL", phase);
+            for(size_t i = 0; i < statsList.size(); i++)
+            {
+                perfStatsList.push_back(statsList[i]);
+            }
+
+            statsList = _tileLatency.GetLatencyStats("TL", phase);
+            for(size_t i = 0; i < statsList.size(); i++)
+            {
+                perfStatsList.push_back(statsList[i]);
+            }
         }
+    }
 
-        file << runMs << "\n";
+    PerfMetricInfo GetStressStats(size_t runMs, std::string testPhase) //need to figure out what phase it is
+    {
+        PerfMetricInfo stressStats = {testPhase, "Stress run (ms)", runMs};
+        return  stressStats;
     }
 
-    void dumpNetworkStatsToCSV(size_t recievedKbs, size_t sentKbs)
+    PerfMetricInfo GetCPUUSageStats(size_t cpuUsage, std::string testPhase)
     {
-        std::ofstream file("Network.csv", std::ios::out | std::ios::app);
+        PerfMetricInfo cpuStats = {testPhase, "CPU Usage (us)", cpuUsage};
+        return cpuStats;
+    }
+
+    std::vector<PerfMetricInfo> GetNetworkStats(size_t recievedKb, size_t sentKb, std::string testPhase)
+    {
+        std::vector<PerfMetricInfo> networkStatsList;
+
+        networkStatsList.push_back(PerfMetricInfo{testPhase, "Bytes recieved (kB)",  recievedKb});
+        networkStatsList.push_back(PerfMetricInfo{testPhase,"Bytes sent (kB)", sentKb});
```

Comment:
one space after comma, not none, not two ;-)
Please read your diff carefully before committing.